### DISCUSSION TITLE
LFS - how many CPUs are used by a process when CPUs are specified for a job?

### DISCUSSION THREAD

**Discussion by complexgenome** [nextflow-io/nextflow#4164] [Q&A] [answered]
I run GATK tool in different processes. I've following config file that defines job memory, wall time and such. At the same time, process has memory specified along with CPU

```
profiles {
	minerva {
		process.executor = 'LSF'
		process.queue = 'premium'
        process.memory = '100GB'
		process.cpus = 3
		process.time = '50 h'
		process.clusterOptions =  ' -P acc_pareks02a  -R affinity[core(12)] '
	}
}

	withName: gatk_markduplicates {
        publishDir = [
		[
            	path: "${params.outdir}/gatk_mark_duplicates/normal/",
            	mode: 'copy',
		pattern:'*_N*'
		],
		[
            	path:"${params.outdir}/gatk_mark_duplicates/tumor/",
            	mode:'copy',
		pattern: '*_T*'
		]
        ]
	memory='40 GB'
	cpus=2
	}

```
Here I've `gatk_markduplicates` 

My question is how many CPUs are requested for **job** with gatk mark duplicates process: 2 or 3? 
In my limited understanding the job would have 3 CPUs but the process 2.

**Comment by bentsherman** [comment, marked as answer]
The priority ranking is documented [here](https://nextflow.io/docs/edge/config.html#selector-priority). There is no separation between process and job, a job is an instance of a process,

**Comment by complexgenome** [comment]
Thanks for your reply. It helps
